opnode/rollup/driver/state.go

```diff
@@ -196,13 +212,25 @@ func (s *state) createNewL2Block(ctx context.Context) (eth.L1BlockRef, error) {
 		s.log.Error("Error finding next L1 Origin", "err", err)
 		return eth.L1BlockRef{}, err
 	}
-	if nextOrigin.Time <= s.Config.BlockTime+s.l2Head.Time {
-		s.log.Trace("Skipping block production because the next block time is behind the next L1 Origin", "l2Head", s.l2Head, "l1Origin", nextOrigin)
+	nextL2Time := s.l2Head.Time + s.Config.BlockTime
+	// If we are behind the next L1 origin, then we should be deriving empty blocks (regular verifier work) until the L2 time catches up with L1 again.
+	if nextL2Time < nextOrigin.Time {
+		s.log.Warn("Skipping block production because the next block time is behind the next L1 Origin",
+			"l2Head", s.l2Head, "nextL2Time", nextL2Time, "l1Origin", nextOrigin, "l1OriginTime", nextOrigin.Time)
+		return eth.L1BlockRef{}, nil
+	}
+
+	// TODO: we can give ourselves less slack here, and prefer to adopt the next L1 block more eagerly,
+	// to ensure we have more time to submit the next block we produce, at the cost of sequencing less eagerly.
+	maxL2Time := nextOrigin.Time + s.Config.MaxSequencerDrift
```

Comment:
Ditto need at least one L2 block per epoch.

Reply:
It's a duplicate with the check in the `createNewBlock` method (this function is the only caller). I'll fix one and remove the other.